### PR TITLE
Add FromSql / ToSql for smol_str::SmolStr.

### PR DESCRIPTION
**Pull request by BratSinot**

### CODE REVIEW

File: postgres-types/src/lib.rs [hunk 1]
```diff
@@ -629,6 +630,18 @@ impl<'a> FromSql<'a> for Box<str> {
     }
 }
 
+#[cfg(feature = "smol_str")]
+impl<'a> FromSql<'a> for smol_str::SmolStr {
```
Comment (sfackler) [inline postgres-types/src/lib.rs:634]: These impls should go in their own file like the other optional types.

File: postgres-types/Cargo.toml [hunk 2]
```diff
@@ -48,3 +48,5 @@ uuid-08 = { version = "0.8", package = "uuid", optional = true }
 uuid-1 = { version = "1.0", package = "uuid", optional = true }
 time-02 = { version = "0.2", package = "time", optional = true }
 time-03 = { version = "0.3", package = "time", default-features = false, optional = true }
+
+smol_str = { version = "0.1.23", default-features = false, optional = true }
```
Comment (sfackler) [inline postgres-types/Cargo.toml:52]: This should be `smol_str-01` = { version = "0.1.23", package = "smol_str", default-features = false, optional = true }`.

**Comment by sfackler** [comment]
Can you also update the docs for FromSql and ToSql?

**Comment by BratSinot** [comment]
> Can you also update the docs for FromSql and ToSql?

Done.

**Comment by sfackler** [comment]
It looks like it needs a `cargo fmt`. If you pull in master the clippy failure should be fixed.

**Comment by BratSinot** [comment]
> It looks like it needs a `cargo fmt`. If you pull in master the clippy failure should be fixed.

Yeah, I saw that clippy warning too =)

**Comment by sfackler** [comment]
Thanks!

**Pull request merged**